src/TailoringWindow.cpp

```diff
@@ -300,160 +265,71 @@ void TailoringWindow::synchronizeProfileItem()
     mProfileItem->setIcon(0, getShareIcon("profile.png"));
 }
 
-void TailoringWindow::synchronizeTreeItem(QTreeWidgetItem* treeItem, struct xccdf_item* xccdfItem, bool recursive)
+void TailoringWindow::synchronizeTreeItem()
 {
-    ++mSynchronizeItemLock;
+    // setting visible as false seems to be the only way to avoid entering the "if" in
+    // https://github.com/qt/qt/blob/4.8/src/gui/itemviews/qabstractitemview.cpp#L3190
+    mUI.itemsTree->setVisible(false);
+    synchronizeTreeItemSelections(mBenchmarkItem);
+    mUI.itemsTree->setVisible(true);
```

Comment:
Please, make it clearer that we are making it invisible and visible again due to perceived slowness when changing state of long trees.